### PR TITLE
Add handling for workbooks in personal spaces which will not have pro…

### PR DESCRIPTION
**Pull request by bcantoni**
WIP testing a fix for #873

### CODE REVIEW

File: tableauserverclient/models/workbook_item.py
Comment (jacalata) [inline tableauserverclient/models/workbook_item.py:280]: So this just inserts a meaningless project id to avoid code crashing on the empty value, is that right? Is it possible to put in a human-readable string that makes this more obvious, instead of a uuid?

**Comment by t8y8** [comment]
Does the REST API return anything for the container type, I know in the metadata API we say "containerType" and "containerName" -- it would be nicer if we just modeled that rather than had fake strings.

We could also just make project_id not required on the model and allow it to be none, then handle that in the Serializer.

Fake rando strings feels like my 3rd choice.

(Side note, I seem to have lost my commit bit, was that intentional?)

File: tableauserverclient/models/workbook_item.py
Comment (bcantoni) [reply tableauserverclient/models/workbook_item.py:280]: Yes that's right. And upon further thought this is not a great way to solve it, especially because it will populate project_id with something that _looks like_ a project ID but really isn't.

**Comment by bcantoni** [comment]
@t8y8 I don't see anything in the REST response like that. Instead maybe it's simpler to change the class instantiation from using project_id to just the workbook id (which should always be there and should be unique).

Also I'm not sure where you were dropped, it may have been from team membership changes. In any case I've added you back here with permissions.

**Comment by t8y8** [comment]
> @t8y8 I don't see anything in the REST response like that. Instead maybe it's simpler to change the class instantiation from using project_id to just the workbook id (which should always be there and should be unique).

Yeah, that seems reasonable, and it cleans up `WorkbookItem` a bit. The lowest impact change it to make `project_id` be `project_id=None` in the signature, and then handle it later down in WorkbookRequest serializers, if it's none don't add a Project element. A larger refactor is to get rid of project_id all together, or move it as a keyword-only argument, and the new way to set a project would just be `workbook.projectI_id = ProjectItem or id string`. I think both are fine approaches, one is riskier.

On the side, it's probably worth pinging whatever team owns that now (Personal Spaces, I guess) because it's a miss, imo, that the REST API doesn't include the information. We could update WorkbookItem to handle personal spaces at a later time.

**Comment by matthieucan** [comment]
Hello,
I believe this is a fix for a bug that is hitting our prod since ~1 week. We are unable to run any of our Tableau-related jobs, which all try to retrieve lists of workbooks.
Would it be possible to provide us with an ETA for a resolution and a bugfix release?
Many thanks in advance!

**Comment by ChristophGmeiner** [comment]
> Hello,
> I believe this is a fix for a bug that is hitting our prod since ~1 week. We are unable to run any of our Tableau-related jobs, which all try to retrieve lists of workbooks.
> Would it be possible to provide us with an ETA for a resolution and a bugfix release?
> Many thanks in advance!

Same here!

**Comment by jacalata** [comment]
The fake uuid is not ideal but will definitely get people unblocked here, so worth going ahead with.

**Comment by matthieucan** [comment]
> The fake uuid is not ideal but will definitely get people unblocked here, so worth going ahead with.

Sounds good, when can we expect it?

**Comment by adeturckheim** [comment]
Hi there, anything blocking this merge request? That's a blocker for us

**Review by jacalata — APPROVED** [review]

**Comment by matthieucan** [comment]
Hi @jacalata,
Thank you for merging :pray: Could you give an estimate on when the next release including this fix is likely to happen?

**Comment by alexandruandrei90** [comment]
Hi, thank you for merging the change - do you know when this will be released?

**Comment by jacalata** [comment]
It should be this week.

**Comment by alexandruandrei90** [comment]
Hi, are there any updates on the release?

**Pull request merged**
